crates/models/bloom/src/lib.rs

```diff
@@ -45,6 +45,15 @@ pub struct Bloom {
 
 unsafe impl Send for Bloom {}
 unsafe impl Sync for Bloom {}
+impl Clone for Bloom {
+    fn clone(&self) -> Bloom {
+        // Clone the fields here
+        Bloom {
+            // field1: self.field1,
```

Comment:
Yeah, this was about as far as I got when I started implementing it myself 😅 

The tensors need to be cloned into the new context. We should define a method for cloning a tensor from another context.